### PR TITLE
Deprecate templates and start using templates from SonataAdminBundle

### PR DESCRIPTION
**Pull request by kunicmarko20**
I am targeting this branch, because I am deprecating templates from this bundle and switching to templates in `SonataAdminBundle`

## Changelog

```markdown

### Changed
- Switched to templates from SonataAdminBundle
### Deprecated
- Association templates
```

## To do

- [x] Wait for merge of https://github.com/sonata-project/SonataCoreBundle/pull/495
- [x] Wait for new release of [SonataAdminBundle](https://github.com/sonata-project/SonataAdminBundle) to bump composer version
- [x] Wait for new release of [SonataCoreBundle](https://github.com/sonata-project/SonataCoreBundle) to bump composer version
- [ ] Test if this breaks something

## Subject

As asked from me [here](https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/pull/256#issuecomment-350207607) I am helping this bundle start using templates from `SonataAdminBundle` as mentioned already I do not use this bundle so please test if this breaks something. One more thing, [here](https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/blob/3.x/src/Builder/ShowBuilder.php#L115) we are including template that [does not exist](https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/tree/3.x/src/Resources/views/CRUD) so I was not sure what to do about it?

### CODE REVIEW

**Comment by jordisala1991** [comment]
About the template that does not exist, does that template exist on the admin bundle? If yes, use the admin bundle one, if not we need to check more why was this condition if is not working

**Comment by kunicmarko20** [comment]
Based on ListBuilder I guess this is the corresponding template for ShowBuilder.

File: src/Resources/views/CRUD/edit_mongo_association_script.html.twig [hunk 1]
```diff
@@ -16,6 +16,8 @@ This code manage the many-to-[one|many] association field popup
 
 #}
 
+{% sonata_template_deprecate '@SonataAdminBundle/CRUD/Association/edit_many_script.html.twig' %}
```
Comment (core23) [inline src/Resources/views/CRUD/edit_mongo_association_script.html.twig]: Do we still need the `Bundle` part in `SonataAdminBundle` when refering to the new twig template path syntax?

Comment (kunicmarko20) [reply src/Resources/views/CRUD/edit_mongo_association_script.html.twig]: @core23 I think we do not need it, did anyone test if this transition break something?

**Comment by kunicmarko20** [comment]
Removed `Bundle` from namespace when referencing new template name, did someone have time to test this?

File: src/Builder/ListBuilder.php [hunk 2]
```diff
@@ -121,9 +121,9 @@ public function fixFieldDescription(AdminInterface $admin, FieldDescriptionInter
 
             if (null === $template) {
                 if (ClassMetadataInfo::ONE == $fieldDescription->getMappingType()) {
-                    $template = 'SonataDoctrineMongoDBAdminBundle:CRUD:list_mongo_one.html.twig';
+                    $template = 'SonataAdminBundle:CRUD/Association:list_many_to_one.html.twig';
```
Comment (core23) [inline src/Builder/ListBuilder.php]: Shouldn't you need the `@` prefix?

Comment (kunicmarko20) [reply src/Builder/ListBuilder.php]: @core23 no as this is still the way with `:` instead of `@`.

Comment (kunicmarko20) [reply src/Builder/ListBuilder.php]: should that be changed also?

Comment (core23) [reply src/Builder/ListBuilder.php]: Would be nice, if we use one way for alle template paths (new or old one).

Also we need a a twig dependency in `composer.json`.

**Comment by kunicmarko20** [comment]
Well if the point is to use only one way then we should stick with `:` as that is used all over bundle or switch to `@`?

File: src/Builder/ListBuilder.php (hunk 2 above)
Comment (kunicmarko20) [reply src/Builder/ListBuilder.php]: @core23 Well if the point is to use only one way then we should stick with `:` as that is used all over bundle or switch to `@`?

Comment (kunicmarko20) [reply src/Builder/ListBuilder.php]: also switching to `@` should probably a different PR?

Comment (jordisala1991) [reply src/Builder/ListBuilder.php]: Doing it here would be nice if you can :)

Comment (kunicmarko20) [reply src/Builder/ListBuilder.php]: Replace all templates paths in bundle with `@` ? @jordisala1991

Comment (jordisala1991) [reply src/Builder/ListBuilder.php]: Not all, but the ones that you are changing yes

**Comment by enleur** [comment]
I guess it closes #192?

**Comment by kunicmarko20** [comment]
This should finally be done, could you review it one more time?

**Comment by jordisala1991** [comment]
Please review @sonata-project/contributors

File: src/Resources/views/CRUD/show_mongo_many.html.twig [hunk 3]
```diff
@@ -12,6 +12,8 @@ file that was distributed with this source code.
 
 {% extends 'SonataAdminBundle:CRUD:base_show_field.html.twig' %}
```
[the remark this answers is not in the document]
Comment (jordisala1991) [reply src/Resources/views/CRUD/show_mongo_many.html.twig]: This template is deprecated now. IMO we should not spend more time with them

**Comment by enleur** [comment]
Reference to https://github.com/sonata-project/dev-kit/issues/374

**Comment by greg0ire** [comment]
Please fix conflicts

**Comment by jordisala1991** [comment]
Can you do a rebase instead of a regular merge to fix conflicts? do you know how to do it?

**Comment by kunicmarko20** [comment]
You are fast, just did it :smile:

**Comment by jordisala1991** [comment]
Let's do it.. If someone finds something that breaks when using the sonataAdmin templates we will fix it then

**Comment by jordisala1991** [comment]
Thank you @kunicmarko20

**Pull request merged**